internal/pkg/service/common/etcdop/watch.go

```diff
@@ -0,0 +1,121 @@
+package etcdop
+
+import (
+	"context"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type EventType int32
+
+const (
+	CreateEvent EventType = iota
+	UpdateEvent
+	DeleteEvent
+)
+
+type EventT[T any] struct {
+	op.KeyValueT[T]
+	Type   EventType
+	Header *etcdserverpb.ResponseHeader
+}
+
+func (e *EventT[T]) Rev() int64 {
+	return e.Header.Revision
+}
+
+func (v Prefix) Watch(ctx context.Context, client *etcd.Client, opts ...etcd.OpOption) etcd.WatchChan {
+	opts = append([]etcd.OpOption{etcd.WithPrefix()}, opts...)
+	return client.Watch(ctx, v.Prefix(), opts...)
+}
+
+func (v PrefixT[T]) Watch(ctx context.Context, client *etcd.Client, handleErr func(err error), opts ...etcd.OpOption) <-chan EventT[T] {
+	typedCh := make(chan EventT[T])
+	v.doWatch(ctx, client, handleErr, typedCh, opts...)
+	return typedCh
+}
+
+func (v PrefixT[T]) GetAllAndWatch(ctx context.Context, client *etcd.Client, handleErr func(err error), opts ...etcd.OpOption) <-chan EventT[T] {
```

Comment:
K tomu `GetAllAndWatch` sa hodi napisat este poznamku, ze rovnako dobre by mohlo fungovat `Watch(WithRev(0), WithCreatedNotify())`, tj. ze chces vsetky hodnoty, aj tie co uz existuju.

Takzko poveda co je lepsie:
- ci pouzit na ten initial load tiez watcher, a nechcet si to streamovat po jednom
- alebo pouzit iterator, ako som to spravil, a nacitavat to po 100-vkach.
- ked bude cas, tak spravit benchmark, ... ale teraz je to jedno, interface tejto metody sa nezmeni

Reply:
Hmm, zajímavý nápad, to jsem si nevšiml že jde ty "notifikace" dostat i takto zpětně. 👍 Ale je pravda že to asi může na začátku docela trvat když tam těch dat bude hodně.